src/main/java/net/atos/entng/calendar/helpers/EventHelper.java

```diff
@@ -281,118 +288,202 @@ public void handle(Either<String, JsonObject> event) {
+                                        proceedOnUserList(listOfUsers, calendar, handler);
                                     }
+                                } else {
+                                    handler.handle(null);
                                 }
-                            }); // end neo4j.execute
-                        } // end if (!shareIds.isEmpty())
-                        else {
-                            handler.handle(null);
-                        }
-                    } // end if shared != null
-                } // end  if (event.isRight())
-            } // end handle
+                            }
+
+                        }); // end neo4j.execute
+                    } // end if (!shareIds.isEmpty())
+                    else {
+                        handler.handle(null);
+                    }
+                } // end if shared != null
+            } // end  if (event.isRight())
         });
     }
 
+    /**
+     * Prepare notification user list so that it contains only the people the event is shared with and that have access to the calendar,
+     * and the calendar owner and the event owner if they are different from the person editing
+     *
+     * @param listOfUsers the users with access to the calendar
+     * @param user the user that edited the event
+     * @param calendar the targeted calendar
+     * @param calendarEvent the edited event
+     * @param handler
+     */
+    private void restrictListOfUsers(JsonArray listOfUsers, final UserInfos user, JsonObject calendar, JsonObject calendarEvent,
+                                     final Handler<Map<String, Object>> handler) {
+
+        //make array with userIds & groupIds from event
+        List<String> calendarEventShared = getSharedIds(calendarEvent.getJsonArray("shared", new JsonArray()));
+
+        //get all userIds from groups for event
+        userService.fetchUser(calendarEventShared, user)
+            .onSuccess(e -> {
+                List<String> calendarEventShareIds = e.stream().map(User::id).collect(Collectors.toList());
+
+                //keep ids from shareIds that appear in calendarEventShareIds
+                List<String> userIds = ((List<JsonObject>) listOfUsers.getList())
+                        .stream()
+                        .map((currentUser) -> currentUser.getString("id"))
+                        .collect(Collectors.toList());
+                userIds.retainAll(calendarEventShareIds);
+
+                JsonArray finalUserIds = new JsonArray();
+
+                if (!user.getUserId().equals(calendar.getJsonObject("owner").getString("userId"))) {
+                    userIds.add(calendar.getJsonObject("owner").getString("userId"));
+                }
+                if (!user.getUserId().equals(calendarEvent.getJsonObject("owner").getString("userId"))){
+                    userIds.add(calendarEvent.getJsonObject("owner").getString("userId"));
+                }
+                userIds.stream()
```

Comment:
TU n'as pas un warning "Result of 'Stream.collect()' is ignored " ? tu n'utilises pas le résultat ou autre